### DISCUSSION TITLE
Running a variable number of steps, concurrently

### DISCUSSION THREAD

**Discussion by hermanhobnob** [argoproj/argo-workflows#6683] [Q&A] [answered]
I've got a template defining a DAG and I'd like to be able to pass in a parameter saying how many copies of one of the tasks I want to run in parallel.

I thought I could achieve this by adding a withParam to the tasks I want to run multiple copies of:

```
withParam: "{{=sprig.toJson (sprig.until inputs.parameters['number-of-tasks'])}}"
```

but I don't believe the withParam template is getting expanded. Am I missing an obvious way to achieve this?
I can come up with a complete example in due course but I wondered if I was missing something dumb, like maybe the template hasn't expanded simply because I've got a syntax error or somesuch. Is there any way to debug that?

Any suggestions very welcome. Thanks.

**Comment by hermanhobnob** [comment, marked as answer]
Typically, within a couple of hours of posting a question I find the answer myself. The solution is to use

```
withSequence:
  count: "{{inputs.parameters.number-of-tasks}}"
```

works like a charm. Thanks.

**Reply by samene** [reply]
@hermanhobnob can you explain where you found this solution. Because this is not what is mentioned in the docs
https://argoproj.github.io/argo-workflows/variables/#expression

**Reply by hermanhobnob** [reply]
I was reading the source trying to understand how the sprig template stuff might be used and then I spotted this, https://github.com/argoproj/argo-workflows/blob/4c9c922924be2a299995fc06efbaef15c7fb0f84/workflow/controller/steps.go#L444 and I thought, that looks like what I'm actually looking for.

I notice there are a few examples using withSequence, eg https://github.com/argoproj/argo-workflows/blob/4e450e250168e6b4d51a126b784e90b11a0162bc/examples/loops-sequence.yaml

But yeah, I didn't find it by reading the docs.
